src/hotspot/share/code/nmethod.hpp

```diff
@@ -491,6 +497,12 @@ class nmethod : public CodeBlob {
 #endif
   );
 
+  // Relocate the nmethod to the code heap identified by code_blob_type.
+  // Returns nullptr if the code heap does not have enough space or the
+  // nmethod is unrelocatable, otherwise the relocated nmethod.
```

Comment:
... nmethod is unrelocatable, or nmethod is invalidated during relocation, otherwise ...